### DISCUSSION TITLE
List/Table Field

### DISCUSSION THREAD

**Discussion by tobiasiv** [payloadcms/payload#7141] [Feature Requests & Ideas]
### Current Challenge
Currently, displaying "nested" related collection documents in a table view is challenging and requires custom UI components. For example, consider the relationship between **Customers** and **Orders**. Customer "John Doe" is related to multiple orders: Order 1, Order 2, etc.

In e-commerce platforms such as Magento 2 and Medusa.js, it is common practice to show these relationships in tables. This presentation enhances the usability and clarity of relationships and data.

### Proposed Solution
It would be beneficial to have a list/table field especially for relationships, where users can view, edit, and delete related entries directly within the parent document's view. This field could also be used to represent other (non-relationship) data.

### Alternative Presentation
An alternative presentation for relationships could be considered, but it would only make sense if bidirectional relationships already exist. The issue is that since the [Postgres-Relationship-Column-Update](https://github.com/payloadcms/payload/pull/6339) , you want to keep the relationship on the many side to preserve a clear database schema. However, you also want to display the related list of data on the one side not the many side (e.g. customer layout should contain orders, not the other way around).

### Quick Mockup
Here's an example mockup of what a table field could look like (just the list view within the edit view):
![table-field](https://github.com/user-attachments/assets/e10546c2-bda3-4ea2-be65-682c68cc5dce)

### Magento 2
![magento-2](https://github.com/user-attachments/assets/c24376ec-2f7c-4a63-b619-742b6186cd7a)

### Medusa.js
![medusa-js](https://github.com/user-attachments/assets/f2fac249-0c30-4fbe-9652-0e4d44b3727b)

Implementing this feature would significantly enhance the usability and functionality of our system, aligning it with common practices in other e-commerce platforms.

**Comment by gabrielmedel** [comment]
Yes, it would be definitely good for the payload CMS ecosystem. It could be related to a collection or be inside of the collection using JSON field. I vote to use TansTack Table for this.

Also I saw that someone made a field: [https://github.com/notchris/payload-table-field/](url)

But is on the V2 version. I tried to migrate it to v3 but many props and components have changed.

**Comment by tobiaghiraldini** [comment]
I was having the same need and I am trying to create a custom field for my relationships, but with no luck yet. there are so many undocumented things to know about and my implementation is always lacking something not trivial to work as expected.
it would be great that at least the way to create custom fields was a little bit clearer and straightforward , at least for basic needs.
for what it's worth, I tried approaching this need creating a custom relationship field, then tried to customize its Field and Cell components, but something is always erroring, and I couldn't reach some point where I know my way forward. I'm always lost in payload quirks

**Comment by thddrew** [comment]
@gabrielmedel @tobiaghiraldini 

Does the [JOIN](https://payloadcms.com/docs/fields/join) field work for you guys? I found this works in my case.
For example, showing all of the customer's bookings in a Customer view.

**Reply by gabrielmedel** [reply]
The Join field is something similar. But my usecase is to have a bulk edit table like spreadsheets for some relationship. So it will be more of a view type for a join field. Maybe a custom component for the join field but I would need a minimum example of a basic join field component customization
